apps/monarch/monarch-frontend/src/app/components/DistanceFilter/index.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState, useEffect, useContext } from 'react';
+import {
+    Box,
+    FormControl,
+    FormLabel,
+    Switch,
+    Heading,
+    InputGroup,
+    InputRightElement,
+    Input,
+    Slider,
+    SliderMark,
+    SliderTrack,
+    SliderThumb,
+    SliderFilledTrack,
+    Tooltip,
+} from '@chakra-ui/react';
+import { CheckIcon } from '@chakra-ui/icons';
+import { controller } from '../../actionsController';
+import { QueryContext } from '../../SearchTherapists';
+
+const isValidZipcode = (zipcode: string): boolean => {
+    return /(^\d{5}$)|(^\d{5}-\d{4}$)/.test(zipcode);
+};
+
+const defaultMaxDistance = 100;
+const defaultZipcode = "";
+
+const DistanceFilter: React.FC = () => {
+    const [toggleFilter, setToggleFilter] = useState(false);
+    const [toggleToolTip, setToggleToolTip] = useState(false);
+    const [maxDistance, setMaxDistance] = useState<number>(defaultMaxDistance);
+    const [zipcode, setZipcode] = useState<string>(defaultZipcode);
+
+    const queryContext = useContext(QueryContext);
+
+    useEffect(() => {
+        if (toggleFilter && isValidZipcode(zipcode)) {
+            queryContext?.setDistanceFilterEnabled(true);
+            controller.extractGeocodeFromZipcode(zipcode).then(queryContext?.setClientCoordinates);
+        } else {
+            queryContext?.setDistanceFilterEnabled(false);
+            queryContext?.setClientCoordinates(undefined);
+        }
+
+        if (!toggleFilter) {
+            setZipcode(defaultZipcode);
+            setMaxDistance(defaultMaxDistance);
+            queryContext?.setClientCoordinates(undefined);
+            queryContext?.setSearchQuery({ ...queryContext.searchQuery, maxDistance: defaultMaxDistance});
+        }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I disabled the exhaustive-deps rule for this line because it wanted me to include the queryContext as a dependency. The queryContext contains methods that would lead to an infinite cycle of re-rendering this component (I think).